### PR TITLE
Refactor with dynamic command loading

### LINKED ISSUE #6: Slash Command Handlers
We already have basic slash commands, but it'd be good to make a system in the style of the message handlers

### LINKED ISSUE #8: Dynamic command loading
Each command resides within a file

The file contains: 

- Name of the command as it is called
- Description for the user
- `execute()` method

### PR DESCRIPTION
**Pull request by dislocated-su**
This pr will aim to restructure the project and refactor the code to move commands and logic out of index.ts

Resolves #8
Might resolve #6

### CODE REVIEW

**Comment by dislocated-su** [comment]
The docs will need updating to reflect the changes in this PR

File: src/index.ts [hunk 1]
```diff
@@ -1,144 +1,110 @@
...
+const slashCommands: Collection<ApplicationCommandData, (event: cmdEvent) => any> = new Collection();
+
+import fs from 'fs'
+
+fs.readdir('./dist/commands/message', (err, allFiles) => {
+    if (err) console.log(err);
+
+    let files = allFiles.filter(f => f.split('.').pop() === ('js')); // ignore .js.map files
+
+    if (files.length <= 0) console.log('No message replies found!');
```
Comment (tomkettle1) [inline src/index.ts]: Could reword this cos not all of these will necessarily be for message replies.
(Getting nit picky here lmao)

Comment (dislocated-su) [reply src/index.ts]: Wouldn't anything else be a slash command?

Comment (tomkettle1) [reply src/index.ts]:
```suggestion
    if (files.length <= 0) console.log('No message handlers found!');
```

**Review by tomkettle1 — APPROVED** [review]
All looks good! (Apart from the nitpickiest change lmao)

**Comment by dislocated-su** [comment]
Alright, good to go then.

File: src/index.ts (hunk 1 above)
Comment (tomkettle1) [reply src/index.ts]: The hypothetical I can think of is something like if we wanted to count the occurrences of a word/emote etc.

Comment (tomkettle1) [reply src/index.ts]: Like, I think this would be the best place for something like that to live

**Pull request merged**